.github/workflows/docker-image.yml

```diff
@@ -0,0 +1,28 @@
+name: Docker Image CI
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+jobs:
+
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    
+    - name: Buil Javascript tester
+      run: docker build . --file Dockerfile --target tester --tag "backend:tester_${{github.sha}}"
+    
+    - name: Test Javascript
+      run: docker run --name backend-tester --rm "backend:tester_${{github.sha}}"
+
+    - name: Build Java runner
+      run: docker build . --file Dockerfile --target runner --tag "backend:${{github.sha}}"
+
+    - name: Test Java
+      run: docker run --rm "backend:${{github.sha}}" gradle test
```

Comment:
I'm not 100% sure I understand this correctly.

You are building a docker image. During the docker build the gradle build is executed (including unit tests).

Then we come to this commented line here... you start the docker container and gradle test... what exactly happens here? Are these tests that are not executed during the gradle build?